src/main/java/com/auction/auctionapp/rest/exception/ApiExceptionHandler.java

```diff
@@ -0,0 +1,36 @@
+package com.auction.auctionapp.rest.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+@ControllerAdvice
+public class ApiExceptionHandler {
+
+    @ExceptionHandler(value = {RegisterException.class})
+    public ResponseEntity<Object> handleRegisterException(RegisterException e) {
+        ApiException exception = new ApiException(
+                e.getMessage(),
+                e,
+                HttpStatus.BAD_REQUEST,
+                ZonedDateTime.now(ZoneId.of("Z"))
+        );
+        return new ResponseEntity<>(exception, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(value = {LoginException.class})
+    public ResponseEntity<Object> handleLoginException(LoginException e) {
+        ApiException exception = new ApiException(
+                e.getMessage(),
+                e,
+                HttpStatus.BAD_REQUEST,
+                ZonedDateTime.now(ZoneId.of("Z"))
+        );
+        return new ResponseEntity<>(exception, HttpStatus.BAD_REQUEST);
+    }
```

Comment:
Whenever you are doing a copy-paste operation, think about if there is a better way to do it, either to reuse the code by creating new method, or for this case where ExceptionHandler allows you to define multiple exceptions for it.